src/bdkad/KademliaHandler.cpp

```diff
@@ -264,10 +269,43 @@ namespace bdkad
     {
       printf("ERROR: failed to set value.\n");
       context.setResponseCode(500);
-      context.writeError("Failed", "Failed to set value", bdhttp::ErrorCode::GENERIC_ERROR);
+      context.writeError("Failed", "Failed to publish value", bdhttp::ErrorCode::GENERIC_ERROR);
+      return;
     }
 
     context.writeResponse("true");
+
+    Json::Value copy = json;
+
+    std::thread([copy](){
```

Comment:
We should have a single thread to handle logs instead of creating temporary threads which may exceeds ulimit when there are a lot of announcements.